### PR TITLE
migration: Add case about abort migration

### PR DESCRIPTION
**Pull request by cliping**
VIRT-298383 - [VM migration][domjobabort] abort migration job with wrong API flag

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

 (1/2) type_specific.io-github-autotest-libvirt.migration.async_job.abort_migration_with_wrong_api_flag.abort_postcopy_migration_by_domjobabort_without_postcopy: PASS (316.43 s)
 (2/2) type_specific.io-github-autotest-libvirt.migration.async_job.abort_migration_with_wrong_api_flag.abort_precopy_migration_by_domjobabort_with_postcopy: PASS (269.23 s)

**Review by nanli1 — APPROVED** [review]

File: libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    virsh_migrate_desturi = "qemu+tcp://${migrate_dest_host}/system"
+    status_error = "no"
+    virsh_migrate_dest_state = "running"
+    virsh_migrate_src_state = "shut off"
+    domjobabort_on_src = "yes"
+    variants:
+        - abort_postcopy_migration_by_domjobabort_without_postcopy:
+            virsh_migrate_options = '--live --p2p --verbose --postcopy --timeout 3 --timeout-postcopy --postcopy-bandwidth 10 --bandwidth 10'
+            domjobabort_err_msg = "cannot abort migration in post-copy mode"
+            action_during_mig = '[{"func": "libvirt_network.check_established", "after_event": "iteration: '1'", "func_param": "params"}, {"func": "do_domjobabort", "func_param": "params", "need_sleep_time": "3"}]'
```
Comment (Yingshun) [inline libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg:34]: Is 'libvirt_network.check_established' a specific check for postcopy migration, right?

File: libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        - abort_precopy_migration_by_domjobabort_with_postcopy:
+            domjobabort_err_msg = "current job is not outgoing migration in post-copy mode"
+            virsh_migrate_options = '--live --p2p --verbose --bandwidth 3'
+            abort_with_postcopy = "yes"
+            action_during_mig = '[{"func": "do_domjobabort", "func_param": "params"}, {"func": "set_migrate_speed_to_high", "func_param": "params"}]'
```
Comment (Yingshun) [inline libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg:39]: same as above. why do 'set_migrate_speed_to_high' only for precopy migraiton?

File: libvirt/tests/src/migration/async_job/async_job.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+from provider.migration import base_steps
+
+
+def run(test, params, env):
+    """
+    Test async job
```
Comment (Yingshun) [inline libvirt/tests/src/migration/async_job/async_job.py:6]: I think it's better to add 'with wrong API flag' and the description of manual case. What do you think?

File: provider/migration/migration_base.py [hunk 4]
```diff
@@ -703,11 +703,17 @@ def do_domjobabort(params):
...
+    domjobabort_on_src = "yes" == params.get("domjobabort_on_src", "no")
 
-    if postcopy_options:
-        ret = virsh.domjobabort(vm_name, option="--postcopy", debug=True, uri=dest_uri)
+    if domjobabort_on_src:
+        uri = None
     else:
-        ret = virsh.domjobabort(vm_name, debug=True, uri=dest_uri)
+        uri = dest_uri
+    if postcopy_options or abort_with_postcopy:
```
Comment (Yingshun) [inline provider/migration/migration_base.py:713]: I wonder if we'd better have a variant for 'virsh domjobabort' **options** to cover more cases. what do you think?

**Review by chloerh — APPROVED** [review]

File: provider/migration/migration_base.py (hunk 4 above)
Comment (Yingshun) [reply provider/migration/migration_base.py:713]: @cliping  any comments?

File: libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg (hunk 1 above)
Comment (cliping) [reply libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg]: This case don't need to check 'libvirt_network.check_established'. Remove it. Thanks for your reminder.

File: libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg (hunk 2 above)
Comment (cliping) [reply libvirt/tests/cfg/migration/async_job/abort_migration_with_wrong_api_flag.cfg:39]: Updated for postcopy migration. Thanks.

File: libvirt/tests/src/migration/async_job/async_job.py (hunk 3 above)
Comment (cliping) [reply libvirt/tests/src/migration/async_job/async_job.py]: Updated, thanks.

File: provider/migration/migration_base.py (hunk 4 above)
Comment (cliping) [reply provider/migration/migration_base.py]: Updated, thanks.

**Review by Yingshun — APPROVED** [review]

**Pull request merged**